File: lumina_beauty_final_version/lumina_beauty_backend/LuminaBeauty-Model/src/main/java/pe/edu/pucp/luminaBeauty/Model/Reclamo.java
package pe.edu.pucp.luminaBeauty.Model;

import java.time.LocalDateTime;

public class Reclamo {

    private int id_reclamo;
    private String tipo;
    private String asunto;
    private String descripcion;
    private String estado;
    private String prioridad;
    private LocalDateTime resuelto_en;
    private LocalDateTime fecha_creacion;
    private LocalDateTime fecha_actualizacion;

    private Cliente cliente;
    private Pedido pedido;
    private DetallePedido detallePedido;

    public Reclamo() {
        this.estado = "ABIERTO";
        this.prioridad = "MEDIA";
    }

    public int getId_reclamo() {
        return id_reclamo;
    }

    public void setId_reclamo(int id_reclamo) {
        this.id_reclamo = id_reclamo;
    }

    public String getTipo() {
        return tipo;
    }

    public void setTipo(String tipo) {
        this.tipo = tipo;
    }

    public String getAsunto() {
        return asunto;
    }

    public void setAsunto(String asunto) {
        this.asunto = asunto;
    }

    public String getDescripcion() {
        return descripcion;
    }

    public void setDescripcion(String descripcion) {
        this.descripcion = descripcion;
    }

    public String getEstado() {
        return estado;
    }

    public void setEstado(String estado) {
        this.estado = estado;
    }

    public String getPrioridad() {
        return prioridad;
    }

    public void setPrioridad(String prioridad) {
        this.prioridad = prioridad;
    }

    public LocalDateTime getResuelto_en() {
        return resuelto_en;
    }

    public void setResuelto_en(LocalDateTime resuelto_en) {
        this.resuelto_en = resuelto_en;
    }

    public LocalDateTime getFecha_creacion() {
        return fecha_creacion;
    }

    public void setFecha_creacion(LocalDateTime fecha_creacion) {
        this.fecha_creacion = fecha_creacion;
    }

    public LocalDateTime getFecha_actualizacion() {
        return fecha_actualizacion;
    }

    public void setFecha_actualizacion(LocalDateTime fecha_actualizacion) {
        this.fecha_actualizacion = fecha_actualizacion;
    }

    public Cliente getCliente() {
        return cliente;
    }

    public void setCliente(Cliente cliente) {
        this.cliente = cliente;
    }

    public Pedido getPedido() {
        return pedido;
    }

    public void setPedido(Pedido pedido) {
        this.pedido = pedido;
    }

    public DetallePedido getDetallePedido() {
        return detallePedido;
    }

    public void setDetallePedido(DetallePedido detallePedido) {
        this.detallePedido = detallePedido;
    }
}

File: lumina_beauty_final_version/lumina_beauty_frontend_versionbuenadeadmin/LuminaBeauty/Models/Orders/Reclamo.cs
using System.Text.Json.Serialization;

namespace LuminaBeauty.Models
{
    public class Reclamo
    {
        [JsonPropertyName("id_reclamo")]
        public int IdReclamo { get; set; }

        [JsonPropertyName("tipo")]
        public string Tipo { get; set; } = string.Empty;

        [JsonPropertyName("asunto")]
        public string Asunto { get; set; } = string.Empty;

        [JsonPropertyName("descripcion")]
        public string Descripcion { get; set; } = string.Empty;

        [JsonPropertyName("estado")]
        public string Estado { get; set; } = "ABIERTO";

        [JsonPropertyName("prioridad")]
        public string Prioridad { get; set; } = "MEDIA";

        [JsonPropertyName("area_asignada")]
        public string AreaAsignada { get; set; } = string.Empty;

        [JsonPropertyName("resuelto_en")]
        public DateTime? ResueltoEn { get; set; }

        [JsonPropertyName("fecha_creacion")]
        public DateTime? FechaCreacion { get; set; }

        [JsonPropertyName("fecha_actualizacion")]
        public DateTime? FechaActualizacion { get; set; }

        [JsonPropertyName("cliente")]
        public Cliente? Cliente { get; set; }

        [JsonPropertyName("pedido")]
        public Pedido? Pedido { get; set; }

        public string Codigo => $"REC-{IdReclamo:D4}";
    }
}